lib/sidekiq/cli.rb

```diff
@@ -235,6 +235,10 @@ def boot_system
           end
```

Comment:
The block of code just above here (231-235) exists to force eager loading, which (I assume) is because runtime autoloading and threaded workers didn't get along so well.

Autoloading is now threadsafe, as long as the relevant code is wrapped by `@app.executor`. `@app.reloader` does that internally, but the condition below is skipping the reloader when `cache_classes` is true.